### PR TITLE
fix(missions): use event timestamps for stale-mission detection

### PR DESCRIPTION
**Pull request by Th0rgal**
## Summary
Fixes a false-positive stale-cleanup pattern that auto-closed a long-running mission earlier today (mission \`a9e5af73\`):

The agent had been actively running for 2+ hours — 487 \`tool_call\` / \`tool_result\` events watching a CI build via repeated \`gh run watch\` invocations — but had not produced an intervening \`assistant_message\`. Since \`mission.updated_at\` is bumped only by \`update_mission_history\` (turn-boundary persist) and metadata writes, it stayed at the time of the *first* turn boundary while the agent worked. The 2h scanner saw \`updated_at < (now − 2h)\` and auto-closed the mission, even though events were arriving the whole time.

## Fix
Change \`get_stale_active_missions\` in the SQLite store so a mission is stale only when *both* \`mission.updated_at\` **and** the latest \`mission_events.timestamp\` are older than the cutoff. The inner subquery uses the existing \`idx_events_mission(mission_id, sequence)\` index — \`ORDER BY sequence DESC LIMIT 1\` is an O(log n) seek per active mission. Missions with no events fall back to \`updated_at\` via \`COALESCE\`.

\`mission.updated_at\` keeps its existing semantics ("last metadata or turn change"). Stale detection now tracks real activity.

## Tests
Three SQLite-backed regression tests:
- mission with old \`updated_at\` but a fresh \`tool_call\` event → not stale (the regression)
- mission with both old \`updated_at\` and old events → stale (counterpart)
- mission with fresh \`updated_at\` and zero events → not stale (\`COALESCE\` path)

A \`#[cfg(test)] force_backdate_for_test\` helper on \`SqliteMissionStore\` lets tests drive deterministic clock scenarios without sleeping or stubbing \`Utc::now\`.

## Other stores
Unchanged: in-memory is non-persistent (cleanup loop skips it), and the file store has no events log to query.

## Test plan
- [x] \`cargo test --lib\` (699 passed)
- [x] \`cargo clippy --workspace --all-targets -- -D warnings\` clean
- [x] \`cargo fmt --all --check\` clean
- [ ] After merge + redeploy, monitor a long-running CI-watching mission across the 2h boundary

---

> [!NOTE]
> **Medium Risk**
> Changes the criteria used to auto-close "stale" active missions and adds a new query/index; incorrect logic or timestamp assumptions could prevent cleanup or close missions unexpectedly, though coverage is improved with new regression tests.
> 
> **Overview**
> Fixes false-positive stale cleanup for long-running active missions by updating `get_stale_active_missions` (SQLite) to treat a mission as stale only when both `missions.updated_at` *and* the mission’s newest `mission_events.timestamp` are older than the cutoff (using `MAX(timestamp)` and `COALESCE` for missions with no events).
> 
> Adds a new SQLite index `idx_events_mission_timestamp` (and migration to create it) to keep the new query efficient, plus a test-only `force_backdate_for_test` helper and several regression tests covering active-with-recent-events, truly-idle missions, sequence-vs-timestamp edge cases, and no-event missions. Also updates `.gitignore` to ignore `.claude/` session state.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 8886ae1cd0f996544eeb6e2c04d52e2c16e02e2b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/api/mission_store/sqlite.rs [hunk 1]
```diff
@@ -2617,15 +2649,36 @@ impl MissionStore for SqliteMissionStore {
...
-                     FROM missions
-                     WHERE status = 'active' AND updated_at < ?1",
+                     FROM missions m
+                     WHERE status = 'active'
+                       AND max(
+                             m.updated_at,
+                             COALESCE(
+                               (SELECT timestamp FROM mission_events
+                                WHERE mission_id = m.id
+                                ORDER BY sequence DESC LIMIT 1),
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/mission_store/sqlite.rs]: Fixed in 8886ae1. Switched the per-mission tail subquery to `MAX(timestamp)` so an in-place update on a lower-sequence row (text_delta_latest, etc.) is observed. Added `idx_events_mission_timestamp(mission_id, timestamp DESC)` to keep that query O(log n). New regression test `stale_query_uses_max_timestamp_not_max_sequence` constructs the exact "newer-ts-on-lower-seq" layout the in-place-update path produces.
